src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorageQueue.cs

```diff
@@ -42,20 +44,36 @@ public MsmqSubscriptionStorageQueue(MsmqAddress queueAddress, bool transactionsE
             q.MessageReadPropertyFilter = messageReadPropertyFilter;
         }
 
-        public IEnumerable<Message> GetAllMessages()
+        public IEnumerable<MsmqSubscriptionMessage> GetAllMessages()
         {
-            return q.GetAllMessages();
+            return q.GetAllMessages().Select(m => new MsmqSubscriptionMessage(m));
         }
 
-        public void Send(Message toSend)
+        public string Send(string body, string label)
         {
-            toSend.Formatter = q.Formatter;
+            var toSend = new Message()
+            {
+                Recoverable = true,
+                Formatter = q.Formatter,
+                Body = body,
+                Label = label
+            };
+
             q.Send(toSend, GetTransactionType(transactionsEnabled, dontUseExternalTransaction));
+
+            return toSend.Id;
         }
 
-        public void ReceiveById(string messageId)
+        public void TryReceiveById(string messageId)
         {
-            q.ReceiveById(messageId, GetTransactionType(transactionsEnabled, dontUseExternalTransaction));
+            try
+            {
+                q.ReceiveById(messageId, GetTransactionType(transactionsEnabled, dontUseExternalTransaction));
```

Comment:
Given that this change makes the storage "append only" lets just use MessageQueueTransactionType.None ?